src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -215,13 +215,14 @@ private static Expression CreateArgument(ParameterInfo parameter, FactoryContext
 
             if (parameterCustomAttributes.OfType<IFromRouteMetadata>().FirstOrDefault() is { } routeAttribute)
             {
-                factoryContext.TrackedParameters.Add(parameter.Name, RequestDelegateFactoryConstants.RouteAttribute);
-                if (factoryContext.RouteParameters is { } routeParams && !routeParams.Contains(parameter.Name, StringComparer.OrdinalIgnoreCase))
+                var routeName = routeAttribute.Name ?? parameter.Name;
+                factoryContext.TrackedParameters.Add(routeName, RequestDelegateFactoryConstants.RouteAttribute);
+                if (factoryContext.RouteParameters is { } routeParams && !routeParams.Contains(routeName, StringComparer.OrdinalIgnoreCase))
                 {
-                    throw new InvalidOperationException($"{parameter.Name} is not a route paramter.");
+                    throw new InvalidOperationException($"'{routeName}' is not a route parameter.");
```

Comment:
Good catch!

Reply:
Ooh, I know it's super unrelated, but could I bother you to also fix https://github.com/dotnet/aspnetcore/blob/main/src/Http/Routing/src/Template/TemplatePart.cs#L75 and https://github.com/dotnet/aspnetcore/blob/main/src/Http/Routing/src/Template/TemplatePart.cs#L131. II just searched for this typo since it's common.